src/Text/Pandoc/Writers/HTML.hs

```diff
@@ -1047,12 +1048,20 @@ inlineToHtml opts inline = do
                                               strToHtml "’")
                               DoubleQuote -> (strToHtml "“",
                                               strToHtml "”")
-                        in  if writerHtmlQTags opts
+
+                        in if writerHtmlQTags opts
                                then do
                                  modify $ \st -> st{ stQuotes = True }
-                                 H.q `fmap` inlineListToHtml opts lst
+                                 H.q `fmap` inlineListToHtml opts lst' >>= withAttr
                                else (\x -> leftQuote >> x >> rightQuote)
-                                    `fmap` inlineListToHtml opts lst
+                                    `fmap` inlineListToHtml opts lst'
+                          where
+                            lst' = maybe lst (\cs -> cs ++ tail lst) (snd <$> citeSpan)
+                            withAttr = maybe return (addAttrs opts) (fst <$> citeSpan)
+                            citeSpan = listToMaybe lst >>= (\case
```

Comment:
I'm not so sure about the use of `listToMaybe` here.
I think we should just pattern match for the exact structure `Quoted _ [Span attr xs]` where attr includes cite.
If they have `Quoted _ [Span attr xs, Str "hello"]`, for example, it shouldn't have special handling.  That will allow you to simplify your code here.

Also: why not just use the orginal inline list when `writerHtmlQTags` is not set?  We'll get regular quotes around span tags, that's fine.

Reply:
Hmm yeah that would simplify the code. Thank you for the tip! I will change it.